docs/BR.md

```diff
@@ -1008,6 +1065,66 @@ CAs are permitted to treat a record lookup failure as permission to issue if:
 
 CAs MUST document potential issuances that were prevented by a CAA record in sufficient detail to provide feedback to the CAB Forum on the circumstances, and SHOULD dispatch reports of such issuance requests to the contact(s) stipulated in the CAA iodef record(s), if present. CAs are not expected to support URL schemes in the iodef record other than mailto: or https:.
 
+#### 3.2.2.9 Multi-Perspective Issuance Corroboration
+
+Multi-Perspective Issuance Corroboration attempts to corroborate the Primary Domain Validation Determination and Primary CAA Determination (jointly referred to as "primary determinations") by re-completing the entirety of the domain validation process from multiple additional geographic Network Perspectives before subscriber certificate issuance. This process can improve protection against equally-specific prefix Border Gateway Protocol (BGP) attacks or hijacks.
+
+Network Perspectives are considered distinct when the straight-line distance between the two States, Provinces, or Countries they reside in is at least 500 km.
+
+The network infrastructure providing Internet connectivity to a Network Perspective MAY be administered by the same organization providing the computational services required to operate the Network Perspective. 
+
+Each primary and corroborating Network Perspective MUST independently complete the entirety of the following:
+- the domain validation process used to establish the Primary Domain Validation Determination (including DNS lookups for A records), and
+- the CAA record-checking process used to establish the Primary CAA Determination. 
+
+Results or information obtained from one Network Perspective MUST NOT be reused or cached when performing validation through subsequent Network Perspectives (e.g., different Network Perspectives cannot rely on a shared DNS cache to prevent an adversary with control of traffic from one Network Perspective from poisoning the DNS cache used by other Network Perspectives).
+
+[Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](#3225-authentication-for-an-ip-address) describe the validation methods that require the use of Multi-Perspective Issuance Corroboration and how a Network Perspective can corroborate the Primary Domain Validation Determination and the Primary CAA Determination. A CA MAY use corroborating evidence for CAA record compliance for a period equal to the period of validity for Validation of Domain Authorization or Control specified in [Section 4.2.1](#421-performing-identification-and-authentication-functions).
```

Comment:
RFC 8659 states:
> The search for a CAA RRset climbs the DNS name tree from the specified label up to, but not including, the DNS root "." until a CAA RRset is found.

This means that for `x.y.z.example.com` the following zones are checked for CAA records, until a record is found.

- x.y.z.example.com
- y.z.example.com
- z.example.com
- example.com
- com

If the domain `example.com` is added and used for domain control validation (including multi-perspective CAA checks), the attacker can still bypass the multi-perspective CAA check by faking the any of the more specific zones with a new CAA record, for example on `x.y.z.example.com` unless multi perspective CAA checking is done at the time of issuance.

The other challenge might be that at the time of domain control validation the context of the certificate might not be known, so this would require checking for `issue` and `issuewild` and potentially other CAA properties and to record the type of certificates that this domain would be authorized for.

The suggestion was to perform multi perspective CAA checking at the time of domain validation control to avoid the performance impact, but it's the question if this really helps as there are different FQDNs and types of certificates to consider that might be unknown at the time of performing domain control validation.

Reply:
Thanks for bringing this up. As I understand this comment there are largely see two factors worth considering:

1: at time of domain control validation, the exact subdomain may not be known, so a CAA recheck would have to be performed at issuance time on the subdomain. 

2: at time of domain control validation, the certificate type may not be known so the whether the issue or issuewild field is used cannot be determined.

My first thoughts on these two points is that cases where point 1 is applicable simply can't benefit as much from this CAA check at DV time clause. Potentially if CAA evidence was collected from multiple perspectives for "example.com" and a customer requested a certificate for sub.example.com, the CAA evidence for example.com could be reused preventing the need to query recursively up the DNS graph from multiple perspectives. However, in my opinion, as @vanbroup mentioned, a query for sub.example.com in CAA would have to be done at time of issuance.

Point 2 is interesting as well, and I could see it becoming more involved if CAA record types grow in complexity. My personal thoughts on this are that the certificate type concern is not too big of an issue. First, there may be some more advanced APIs, but my general understanding of DNS is that a question like "example.com IN CAA" will always return the entire CAA DNS record set (having knowledge of the entire record set is also required in case there are critical flag records). Thus, whenever a CA makes this query, it will retrieve CAA records with both the "issue" and "issuewild" tags. With this in mind, I think the resolution to point 2 is simply to cache the values of the CAA records fetched from multiple vantage points (as opposed to a can issue or can't issue binary flag) and then resolve the issuance status based on the cached values and certificate type at issuance time.

Overall I still feel it makes sense to have this performance optimization in there. I understand there are some cases where it can't be realized and CAA records would still have to be checked from multiple perspectives at issuance time, but I feel checking CAA records from multiple perspectives is very important for the security of the PKI and the future growth of CAA records. I think any way to reduce the latency/resource overhead without compromising security is beneficial.

Reply:
I agree that you could cache the CAA records (assuming DNSSec signatures etc. are validated), and you could potentially do additional checks based on likelihood (like including an MP CAA check for www.example.com). 

But this would not come for free and require quite a bit of additional implementation work (code that could fail), and meta data for each domain.

I think it would be good to dive a bit deeper into the potential impact and realistic performance gain. 

Maybe it could be described as that when MP CAA has been performed it can be re-used until the current domain name validation expires. This way you would only need do the MPV CAA checking at issuance time when it has not been performed since the last domain control validation check.

This would allow the CA to do MP CAA checking at domain control validation, but would also allow them to perform it at issuance time and re-use it in both cases.

Reply:
@vanbroup I do appreciate the points about additional code complexity and extra meta data (which can add up at large scale).

I wanted to take a step back from this problem and actually think about its impact on the threat space. I think one important thing to consider is that the two cases being discussed are related to domain validation reuse for subdomains and wildcard certificate types. My understanding of the BRs is that both of these cases require some form of validation of DNS information (not just control of a server running at that domain) which would be checked from multiple vantage points. Given the adversary can demonstrate this level of control over DNS, the adversary likely also has the ability to also spoof CAA records. I think there still is some attack surface left open by not rechecking sub-domain CAA or issuewild records from multiple vantage points (e.g., an adversary could have compromised domain validation by attacking an external channel like email without actually controlling the target domain's DNS records), but I also think this is not a huge threat surface.

@vanbroup can you put in a suggested change with the language you are proposing. I like the idea you had to permit reuse since the last validation and also permit rechecking at issuance time and caching as an option.